rust/lance-encoding/src/data.rs

```diff
@@ -95,15 +157,30 @@ impl NullableDataBlock {
         Self {
             data: Box::new(self.data.borrow_and_clone()),
             nulls: self.nulls.borrow_and_clone(),
+            block_info: self.block_info.clone(),
+            used_encoding: self.used_encoding.clone(),
         }
     }
 
     fn try_clone(&self) -> Result<Self> {
         Ok(Self {
             data: Box::new(self.data.try_clone()?),
             nulls: self.nulls.try_clone()?,
+            block_info: self.block_info.clone(),
+            used_encoding: self.used_encoding.clone(),
         })
     }
+
+    // count_nulls will be handled differently after V2.1
+    pub fn count_nulls(&mut self) -> u64 {
+        let nulls_buf = &self.borrow_and_clone().into_buffers()[0];
+        let boolean_buf = BooleanBuffer::new(nulls_buf.into(), 0, nulls_buf.len() * 8);
```

Comment:
You'll need to use `self.data.num_values()`.

`nulls_buf.len() * 8` is not correct because the last byte may not be complete.

Reply:
removed this method completely.